integrations/gcp/gcp_core/utils.py

```diff
@@ -75,23 +75,32 @@ def parse_latest_resource_from_asset(asset_data: AssetData) -> dict[Any, Any]:
     return max_versioned_resource_data["resource"]
 
 
-def should_use_snake_case() -> bool:
+def should_use_snake_case(matching_resource_configs: list[Any] = []) -> bool:
     """
     Determines whether to use snake_case for field names based on preserve_api_response_case_style config.
 
     Returns:
         bool: True to use snake_case, False to preserve API's original case style
     """
-    selector = get_current_resource_config().selector
+    if matching_resource_configs:
+        resource = next(iter(matching_resource_configs), None)
```

Comment:
What happens when there are more than one config in the array ? See comment in main.py on how to resolve this